src/apprt/gtk/Surface.zig

```diff
@@ -346,6 +346,11 @@ cursor: ?*c.GdkCursor = null,
 /// pass it to GTK.
 title_text: ?[:0]const u8 = null,
 
+/// The title of the surface as reported by the terminal.
```

Comment:
See my comment here: https://github.com/ghostty-org/ghostty/issues/5769#issuecomment-2672676710

But I think we should not be changing the _surface_ title. This is about changing the _tab_ title. If we change the surface title the tab title will change every time that you switch focus to a different split if you have multiple splits in a tab.

Reply:
After discussing this with Mitchell on Discord that's how he wants the feature to work right now. The current implementation seems a little awkward to me but maybe I'd have to take another look. So ignore this comment for now.